src/psbt.cpp

```diff
@@ -210,6 +210,20 @@ bool PSBTInputSigned(PSBTInput& input)
     return !input.final_script_sig.empty() || !input.final_script_witness.IsNull();
 }
 
+void UpdatePSBTOutput(const SigningProvider& provider, PartiallySignedTransaction& psbt, int index)
+{
```

Comment:
I know this is move-only, but I found this functionality pretty inscrutable, and think it could do with some additional commenting. Things that would be helpful:

- explain why the caller is instantiating a `HidingSigningProvider` with `nosign` set to true
- explain what the magic `0` and `1` values are in the call to `ProduceSignature()` 
- explain that `ProduceSignature()` isn't really signing (or if it does, then that signature gets thrown away)

Reply:
Added a bunch of comments. Better?